### PR TITLE
Enabling RGB8 Infrared stream

### PR DESCRIPTION
**Pull request by Arun-Prasad-V**
Updated the default supported format of Infra0 stream to "RS2_FORMAT_RGB8".

If this Infra0 stream is enabled, we can't open infra1 and infra2. But, by default all the streams are enabled in ROS2 wrapper. 

So, had to set the default value of '_enable_infra_' param to '_false_' which will disable infra0 stream during init.

### CODE REVIEW

File: realsense2_camera/src/base_realsense_node.cpp [hunk 1]
```diff
@@ -126,9 +126,6 @@ BaseRealSenseNode::BaseRealSenseNode(rclcpp::Node& node,
     _encoding[1] = sensor_msgs::image_encodings::MONO8; // ROS message type
     _encoding[2] = sensor_msgs::image_encodings::TYPE_16UC1; // ROS message type
     _encoding[3] = sensor_msgs::image_encodings::RGB8; // ROS message type
-    
-    // Infrared stream
-    _format[RS2_STREAM_INFRARED] = RS2_FORMAT_Y8;
```
Comment (Arun-Prasad-V) [inline realsense2_camera/src/base_realsense_node.cpp:131]: This variable is not used anywhere.

File: realsense2_camera/launch/rs_launch.py [hunk 2]
```diff
@@ -35,9 +35,9 @@
                            {'name': 'rgb_camera.profile',           'default': '0,0,0', 'description': 'color image width'},
                            {'name': 'rgb_camera.enable_auto_exposure', 'default': 'true', 'description': 'enable/disable auto exposure for color image'},
                            {'name': 'enable_color',                 'default': 'true', 'description': 'enable color stream'},
+                           {'name': 'enable_infra',                 'default': 'false', 'description': 'enable infra0 stream'},
                            {'name': 'enable_infra1',                'default': 'false', 'description': 'enable infra1 stream'},
                            {'name': 'enable_infra2',                'default': 'false', 'description': 'enable infra2 stream'},
-                           {'name': 'infra_rgb',                    'default': 'false', 'description': 'enable infra2 stream'},
```
Comment (Arun-Prasad-V) [inline realsense2_camera/launch/rs_launch.py:40]: ROS2 wrapper is not taking any parameter named "infra_rgb"

File: realsense2_camera/src/profile_manager.cpp [hunk 3]
```diff
@@ -88,7 +88,14 @@ void ProfilesManager::registerSensorUpdateParam(std::string template_name,
     {
         std::string param_name = applyTemplateName(template_name, sip);
         if (params.find(sip) == params.end())
-            params[sip] = std::make_shared<T>(value);
```
Comment (Arun-Prasad-V) [inline realsense2_camera/src/profile_manager.cpp:91]: Here, by default the "_enable__*" param will be set to 'true' for all video streams (color, depth, infra,1&2).

File: realsense2_camera/src/profile_manager.cpp [hunk 4]
```diff
@@ -198,8 +205,9 @@ VideoProfilesManager::VideoProfilesManager(std::shared_ptr<Parameters> parameter
     _module_name(module_name),
     _force_image_default_qos(force_image_default_qos)
 {
-    _allowed_formats[RS2_STREAM_DEPTH] = RS2_FORMAT_Z16;
-    _allowed_formats[RS2_STREAM_INFRARED] = RS2_FORMAT_Y8;
+    _allowed_formats[{RS2_STREAM_DEPTH, 0}] = RS2_FORMAT_Z16;
+    _allowed_formats[{RS2_STREAM_INFRARED, 0}] = RS2_FORMAT_RGB8;
```
Comment (Nir-Az) [inline realsense2_camera/src/profile_manager.cpp:208]: Can you explain the need for this?
It increase the buffer by ×3

Comment (Arun-Prasad-V) [reply realsense2_camera/src/profile_manager.cpp:208]: In D455 & D415:
Infra 0 stream supports           : RGB8, UYVY, RGBA8, etc.,
Infra 1 & 2 streams supports   : Y8, Y16, etc.,

There are no common formats which supports all 3 infra streams. So, have to declare separate default formats for each stream.

File: realsense2_camera/launch/rs_launch.py [hunk 5]
```diff
@@ -35,9 +35,9 @@
                            {'name': 'rgb_camera.profile',           'default': '0,0,0', 'description': 'color image width'},
                            {'name': 'rgb_camera.enable_auto_exposure', 'default': 'true', 'description': 'enable/disable auto exposure for color image'},
                            {'name': 'enable_color',                 'default': 'true', 'description': 'enable color stream'},
+                           {'name': 'enable_infra',                 'default': 'false', 'description': 'enable infra0 stream'},
```
Comment (SamerKhshiboun) [inline realsense2_camera/launch/rs_launch.py:38]: Please add this parameter to rs_multi_camera_launch.py also.

Comment (Arun-Prasad-V) [reply realsense2_camera/launch/rs_launch.py:38]: Anyway, _rs_multi_camera_launch.py_ takes the params from _rs_launch.py_. So, it is getting added automatically.

**Review by SamerKhshiboun — APPROVED** [review]

**Comment by SamerKhshiboun** [comment]
Tested with D455

**Pull request merged**